src/storage/src/source/postgres.rs

```diff
@@ -607,98 +788,74 @@ impl RowSender {
-        let publication_tables = try_indefinite!(
-            mz_postgres_util::publication_info(&self.connection_config, &self.publication).await
-        );
-
-        let client = try_indefinite!(self.connection_config.clone().connect_replication().await);
-
-        // We're initializing this source so any previously existing slot must be removed and
-        // re-created. Once we have data persistence we will be able to reuse slots across restarts
-        let _ = client
-            .simple_query(&format!("DROP_REPLICATION_SLOT {:?}", &self.slot))
-            .await;
-
-        // Validate publication tables against the state snapshot
-        try_definite!(self.validate_tables(publication_tables));
-
-        // Start a transaction and immediately create a replication slot with the USE SNAPSHOT
-        // directive. This makes the starting point of the slot and the snapshot of the transaction
-        // identical.
-        client
-            .simple_query("BEGIN READ ONLY ISOLATION LEVEL REPEATABLE READ;")
-            .await?;
-
-        let slot_query = format!(
-            r#"CREATE_REPLICATION_SLOT {:?} LOGICAL "pgoutput" USE_SNAPSHOT"#,
-            &self.slot
-        );
-        let slot_row = client
-            .simple_query(&slot_query)
-            .await?
-            .into_iter()
-            .next()
-            .and_then(|msg| match msg {
-                SimpleQueryMessage::Row(row) => Some(row),
-                _ => None,
-            })
-            .ok_or_else(|| {
-                ReplicationError::Indefinite(anyhow!(
-                    "empty result after creating replication slot"
-                ))
-            })?;
-
-        // Store the lsn at which we will need to start the replication stream from
-        let consistent_point = try_indefinite!(slot_row
-            .get("consistent_point")
-            .ok_or_else(|| anyhow!("missing expected column: `consistent_point`")));
-        self.lsn = try_definite!(consistent_point
-            .parse()
-            .or_else(|_| Err(anyhow!("invalid lsn"))));
+/// Parses SQL results that are expected to be a single row into a Rust type
+fn parse_single_row<T: FromStr>(
+    result: &[SimpleQueryMessage],
+    column: &str,
+) -> Result<T, ReplicationError> {
+    let mut rows = result.into_iter().filter_map(|msg| match msg {
+        SimpleQueryMessage::Row(row) => Some(row),
+        _ => None,
+    });
+    match (rows.next(), rows.next()) {
+        (Some(row), None) => row
+            .get(column)
+            .ok_or_else(|| anyhow!("missing expected column: {column}"))
+            .and_then(|col| col.parse().or_else(|_| Err(anyhow!("invalid data"))))
+            .err_indefinite(),
+        (None, None) => Err(anyhow!("empty result")).err_indefinite(),
+        _ => Err(anyhow!("ambiguous result, more than one row")).err_indefinite(),
+    }
+}
 
+/// Creates the replication slot and produces the initial snapshot of the data
```

Comment:
I think all of this rustdoc is now outdated? Also: is it true that this "automagically" uses the temporary SLOT that we create right before calling this method, and it therefore produces the data as of `snapshot_lsn`?

Reply:
Updated! This method will run in whatever client context it is invoked with. Depending on the situation call it either in a transaction whose LSN was set by a temp slot or by the actual real slot. You can see the logic in the call site